### PR TITLE
Simplify configs, add docs

### PR DESCRIPTION
**Pull request by Denchick**

### CODE REVIEW

File: balancer/pkg/client.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func NewBalancerClient(conn net.Conn) *BalancerClient {
 	}
 }
 
-func (c BalancerClient) Usr() string {
+func (c BalancerClient) User() string {
```
Comment (reshke) [inline balancer/pkg/client.go:25]: unneeded

File: cmd/balancer/main.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ var rootCmd = &cobra.Command{
 			return fmt.Errorf("error while creating balancer app: %s", err)
 		}
 
-		err = app.ProcBalancer(ctx)
+		err = app.ServeBalancer(ctx)
```
Comment (reshke) [inline cmd/balancer/main.go:45]: unneeded

File: router/pkg/rrouter/manager.go [hunk 3]
```diff
@@ -150,15 +151,15 @@ func NewSessConnManager() *SessConnManager {
 }
 
 func MatchConnectionPooler(client client.RouterClient) (ConnManager, error) {
-	switch client.Rule().PoolingMode {
```
Comment (reshke) [inline router/pkg/rrouter/manager.go:153]: why do we need to rename this thing?

**Review by EinKrebs — APPROVED** [review]

**Review by reshke — CHANGES_REQUESTED** [review]

File: config-example/router.yaml [hunk 4]
```diff
@@ -1,67 +0,0 @@
...
-proto: 'tcp6'
-http_addr: '[::1]:7001'
-world_http_addr: '[::1]:6432'
-
-init_sql: "./test-init.sql"
-
-qrouter:
-    qrouter_type: PROXY
-rules:
-    tls:
```
Comment (reshke) [inline config-example/router.yaml:12]: why do we delete this

File: doc/router/qr/qrouter.md [hunk 5]
```diff
@@ -1,8 +0,0 @@
-
-
-## Single shard configuration
-
-
-## Router mode
```
[the remark this answers is not in the document]
Comment (Denchick) [reply doc/router/qr/qrouter.md:6]: move to docs/ folder

File: pkg/conn/instance.go [hunk 6]
```diff
@@ -76,31 +74,20 @@ func (pgi *PostgreSQLInstance) Receive() (pgproto3.BackendMessage, error) {
...
-func (pgi *PostgreSQLInstance) connect(addr, proto string) (net.Conn, error) {
-	if proto == "" {
-		return net.Dial(defaultProto, addr)
+	netconn, err := net.Dial("tcp", host)
+	if err != nil {
+		return nil, err
 	}
 
-	return net.Dial(proto, addr)
-}
```
Comment (reshke) [inline pkg/conn/instance.go:87]: do we need this changes?

File: router/app/app.go [hunk 7]
```diff
@@ -24,52 +24,49 @@ func NewApp(sg *router.RouterImpl) *App {
 	}
 }
 
-func (app *App) ProcPG(ctx context.Context) error {
-	proto, addr := config.RouterConfig().Proto, config.RouterConfig().Addr
-
-	listener, err := reuse.Listen(proto, addr)
+func (app *App) ServeRouter(ctx context.Context) error {
+	address := net.JoinHostPort(config.RouterConfig().Host, config.RouterConfig().RouterPort)
+	listener, err := reuse.Listen("tcp", address)
```
Comment (reshke) [inline router/app/app.go:29]: we need to respect `proto` setting here

File: router/app/app.go [hunk 8]
```diff
@@ -24,52 +24,49 @@ func NewApp(sg *router.RouterImpl) *App {
...
 	if err != nil {
 		return err
 	}
 	defer func(listener net.Listener) {
 		_ = listener.Close()
 	}(listener)
 
-	spqrlog.Logger.Printf(spqrlog.INFO, "ProcPG listening %s by %s", addr, proto)
+	spqrlog.Logger.Printf(spqrlog.INFO, "SPQR Router is ready on %s", address)
```
Comment (reshke) [inline router/app/app.go:37]: this should be message about postgresql proto listening

File: router/pkg/datashard/conn_pool.go [hunk 9]
```diff
@@ -198,28 +199,20 @@ func (s *InstancePoolImpl) Put(shkey kr.ShardKey, sh Shard) error {
 	}
 }
 
-func NewConnPool(mapping map[string]*config.ShardCfg) DBPool {
-	allocator := func(shardKey kr.ShardKey, host string, rule *config.BERule) (Shard, error) {
-		var err error
+func NewConnPool(mapping map[string]*config.Shard) DBPool {
+	allocator := func(shardKey kr.ShardKey, host string, rule *config.BackendRule) (Shard, error) {
 		spqrlog.Logger.Printf(spqrlog.LOG, "acquire new connection to %v", host)
-		var hostCfg *config.InstanceCFG
```
Comment (reshke) [inline router/pkg/datashard/conn_pool.go:205]: why do we change?

File: router/app/app.go (hunk 7 above)
Comment (Denchick) [reply router/app/app.go:29]: tcp includes tcp4 and tcp6, so we decided its fine

File: router/pkg/rrouter/manager.go (hunk 3 above)
Comment (Denchick) [reply router/pkg/rrouter/manager.go:153]: Decided `pool_mode`

File: pkg/conn/instance.go (hunk 6 above)
Comment (Denchick) [reply pkg/conn/instance.go:87]: Yes

File: balancer/pkg/client.go (hunk 1 above)
Comment (Denchick) [reply balancer/pkg/client.go:25]: Decided
user -> usr
password -> pwd

**Review by reshke — APPROVED** [review]

**Pull request merged**